src/CreateDestroy.ts

```diff
@@ -118,7 +118,8 @@ function ready(
     } else if (descriptor.set != null) {
       kind = 'set';
     }
-    const f: Function = descriptor[kind];
+    type AnyFn = (...args: Array<unknown>) => unknown;
+    const f = descriptor[kind] as AnyFn;
```

Comment:
If you're oing to re-use the type then it needs to go into the `types.ts` file.

https://github.com/MatrixAI/js-async-init/blob/staging/src/types.ts